references/plugins/references.go

```diff
@@ -584,6 +611,102 @@ func (ref *MarkdownReference) GenerateReferenceDocs(ctx context.Context, c commo
 	return ref.CreateMarkdown(ctx, caps, baseRefPath, ReferenceSourcePath, pd)
 }
 
+// GenerateReferenceDocsFromLocalFile generates reference docs from local file
+func (ref *MarkdownReference) GenerateReferenceDocsFromLocalFile(ctx context.Context, c common.Args, baseRefPath string, local string) error {
+	lc, err := ParseLocalFile(local)
+	if err != nil {
+		return fmt.Errorf("failed to parse local file: %w", err)
+	}
+
+	if ref.DefinitionName == "" {
+		return fmt.Errorf("failed to generate docs for all capabilityes from local file")
+	} else {
+		setDisplayFormat("markdown")
+		var (
+			description   string
+			sample        string
+			specification string
+		)
+
+		fileName := fmt.Sprintf("%s.md", lc.Name)
+		if _, err := os.Stat(baseRefPath); err != nil && os.IsNotExist(err) {
```

Comment:
We should call an existed function. If it's not directly callable, refactor it please.